src/tiled.lisp

```diff
@@ -340,16 +347,44 @@
       (setf objects (sort objects #'< :key #'object-y))))
   (values))
 
+(defun %load-template (path resource-loader
+                       &aux
+                         (full-path (uiop:ensure-absolute-pathname path *default-pathname-defaults*))
+                         (data (funcall resource-loader full-path))
+                         (ttemplate
+                          (eswitch ((pathname-type full-path) :test 'string-equal)
+                            ("tx"
+                             (with-input-from-string (stream data)
+                               (parse-xml-template-stream stream full-path)))
+                            ("tj"
+                             (with-input-from-string (stream data)
+                               (parse-json-template-stream stream full-path))))))
+  (let* ((ttileset (ttemplate-tileset ttemplate))
+         (tobject (ttemplate-object ttemplate))
+         (tileset (when ttileset
+                    (%load-external-tileset ttileset 1 resource-loader)))
+         (object (%load-object tobject)))
+    (when tileset
+      (%finalize-object object tobject (list tileset)))
+    object))
+
+(defun load-template (path)
+  (if-let ((template (gethash path *templates-cache*)))
```

Comment:
Since `cl-tiled:load-template` is now a public entry point via the export, we need to bind `cl-tiled::*templates-cache*` here as well.

We'll also need to update line 397 to use `cl-tiled::%load-template`, instead, so that we don't end up overriding the binding while loading a map

Reply:
I'm not sure I get it, the `*templates-cache*` is already bound via `let` to the fresh hash table in `%load-map` at line 231. This way the templates are indeed cached for a single map file, but are loaded from disk anew for another map — that's the single reason for `load-template` wrapper function to exist.

Reply:
Correct, but you made `load-template` public by adding to `:export`, so if a user directly calls `(cl-tiled:load-template ..)` they'll crash on `*templates-cache*` being unbound

Reply:
Ah, my bad, you're totally right.
Unfortunately this way the purpose of the cache will be defeated when using `load-map`, since the template would be loaded from file for every call of `load-object` (which would call `%load-template` and load it past the cache).
I've took the liberty to refactor this bit a little — now the public function `load-template` always loads templates past the cache, and there's new `%load-cached-template` internal fn used by other internals.

Reply:
Oh. I'd hoped to save you the trouble by having recommended you call `%load-template` (instead of `load-template`). But this refactoring works, too

Reply:
Ahh nevermind, I see. If you did that, then `%load-object` would never hit the cache. My bad.
Thanks!